tests/Kernel/RdfSkosGraphServiceTest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\rdf_skos\Kernel;
+
+/**
+ * Tests the RDF SKOS service.
+ */
+class RdfSkosGraphServiceTest extends RdfSkosKernelTestBase {
+
+  /**
+   * Test that op_skos_setup service sets the correct config.
+   */
+  public function testGraphSetupService(): void {
+    // Prepare the graph data.
+    $info = $this->getTestGraphInfo($_ENV['SIMPLETEST_BASE_URL'], 'phpunit');
+    $graphs = [];
+
+    foreach ($info as $name => $item) {
+      $graphs[$name] = $item['uri'];
+    }
+
+    // Call the service to import our data (twice to check for duplicates).
+    for ($i = 0; $i < 2; $i++) {
```

Comment:
I'm not a fan of loops, I think it's ok like it was before, do the add and then

```
$config = \Drupal::service('config.factory')->getEditable('rdf_skos.graphs');
$entity_types = $config->get('entity_types');
```

And then you can do a simple `assertEquals(<expected list>, $entity_types);`
Then call the add with a new array:
```
[
  'fruit' => '...'
  'new_graph' => '...'
]
```
And check do again an assertEquals that like before but we expect only the new_graph to be there.
We can play with fake graph values now, we don't load anymore, just save in config.
Actually we can change the base class of the test too, and make it a simple kernel one! 😍 
So we don't have to `getTestGraphInfo()`, just make up some values.

Reply:
refactored

Reply:
We couldn't use a simple kernel cause we need to setup a lot of stuff. But the rest is done 👍